src/main/webapp/teacher/controlGameInstance/controlGameInstance.js

```diff
@@ -16,54 +16,82 @@
 'use strict';
 
 const db = firebase.firestore();
-let activeGameInstanceId;
+
+// Is triggered when the User logs in or logs out
+function initAuthStateObserver() {
+  firebase.auth().onAuthStateChanged(authStateObserver);
+}
 
 // Triggers when the auth state change for instance when the user signs-in or signs-out.
 function authStateObserver(user) {
   if (user) { // User is signed in!
     // Everything starts working when the User logs in
-    getActiveGameInstanceId(user);
+    loadControlPanel(user);
   } else { // User is signed out!
     console.log("Not logged in");
   }
 }
 
-// Is triggered when the User logs in or logs out
-function initAuthStateObserver() {
-  firebase.auth().onAuthStateChanged(authStateObserver);
+async function loadControlPanel(user) {
+  // Get the Game Instance's ID in which the user is participating
+  const gameInstanceId = await getActiveGameInstanceId(user);
+
+  // Get the Active Game Instance Object from DB
+  const gameInstance = await queryActiveGameInstance(gameInstanceId);
+
+  // Get the Game the GameInstance is using from DB
+  const game = await queryGameDetails(gameInstance.gameId);
+
+  // Add all the GameInstance's Info to UI
+  buildActiveGameInstanceUI({ gameInstanceId, gameInstance, game });
+
+  // Add buttons to control the GameInstance state
+  initUIControlButtons(gameInstanceId);
 }
 
 // Queries the "Users" collection of the DB to get the activeGameInstanceId the User is participating in
+// Returns the gameInstanceId string
 function getActiveGameInstanceId(user) {
   const uid = user.uid;
 
   // Query the User's document in "Users" collection
-  db.collection("users").doc(uid).get().then(function(doc) {
+  return db.collection("users").doc(uid).get().then(function(doc) {
     if (doc.exists) {
       // Get the activeGameInstance's ID in which the User is participating
-      activeGameInstanceId = doc.data().activeGameInstanceId;
-      // Query the info of that GameInstance from the "GameInstance" collections
-      queryActiveGameInstanceDocument(activeGameInstanceId);
+      return doc.data().activeGameInstanceId;
     } else {
+      throw 'User not registered in a GameInstance'
```

Comment:
@brettapeters So for example, there is a possibility that a user accesses the "Play game" page without having registered in any gameInstance previously, so in the loading process of the game, this read will fail and doc.data() will be null. In that case I want to stop everything and tell display in the UI something like "go register first and then come back". At which point of the calling functions should I place a `catch` to abort loading the rest of the UI based on the nonexistent result of this query? In `authStateObserver` which is the function that calls `loadControlPanel`, where this query is called and the result used?